lib/spack/spack/directives.py

```diff
@@ -568,7 +568,7 @@ def _execute_depends_on(pkg):
 
 
 @directive(("extendees", "dependencies"))
-def extends(spec, type=("build", "run"), **kwargs):
+def extends(spec, when=None, type=("build", "run"), patches=None):
```

Comment:
This LGTM, modulo being sure you have seen https://github.com/spack/spack/pull/40326#discussion_r1440312668 If it's ok to break the API, considering that the breakage will probably affect a very small number of users, then this one is mergeable.

Reply:
I think the number is close to if not actually zero -- so I'm in inclined to assume that people have been using kwargs as I've never seen anyone *not* use kwargs for later arguments to directives.